app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -269,34 +284,84 @@ open class ProfilesViewModel(
     _tempProfilePictureBitmap.value = bitmap
   }
 
+  /**
+   * Saves the edited profile to the state.
+   *
+   * @param profile The profile to be saved.
+   */
+  fun saveEditedProfile(profile: Profile) {
+    _editedCurrentProfile.value = profile
+  }
+
   /** Clears the temporary profile picture Bitmap. */
   fun clearTempProfilePictureBitmap() {
     _tempProfilePictureBitmap.value = null
   }
 
+  /** Clears the edited profile from the state. */
+  private fun clearEditedProfile() {
+    _editedCurrentProfile.value = null
+  }
+
+  /** Resets every state related to profile edition. */
+  fun resetProfileEditionState() {
+    clearTempProfilePictureBitmap()
+    clearEditedProfile()
+    _pictureChangeState.value = ProfilePictureState.UNCHANGED
+  }
+
   /**
    * Processes the cropped image by compressing it if it is too large.
    *
    * @param bitmap The cropped Bitmap to be processed.
    */
   fun processCroppedImage(bitmap: Bitmap) {
     _tempProfilePictureBitmap.value = compressImageIfTooLarge(bitmap)
+    _pictureChangeState.value = ProfilePictureState.TO_UPLOAD
+  }
+
+  fun validateProfileChanges(context: Context) {
+    when (_pictureChangeState.value) {
+      ProfilePictureState.TO_UPLOAD ->
+          validateAndUploadProfilePicture(context) { url ->
+            val newProfile = _editedCurrentProfile.value!!.copy(profilePictureUrl = url)
+            updateProfile(newProfile)
+            _selfProfile.value = newProfile
+            // todo: "_selectedProfile.value = newProfile" needs to be removed when all the other
+            // personal profile views are updated to use the selfProfile instead of the
+            // selectedProfile
```

Comment:
It isn't recommended to leave commented code (I have lost points in my M2 individual because i had left commented code). If you want to mark a todo, it is better to write "//TODO:" as it is the current standard used in our codebase

Reply:
That's exactly what I have done. This is not code, this is a comment tagged with `//todo:` to be spotted later on. I also opened an issue to remember to replace `currentProfile` with `selfProfile` in the other user screens.